lib/closex/http_client.ex

```diff
@@ -89,15 +89,47 @@ defmodule Closex.HTTPClient do
 
   # Private stuff...
 
-  defp find(resource, search_term, opts) do
-    opts = merge_search_term_into_opts(search_term, opts)
-
+  defp find(resource, opts) do
     get("/#{resource}/", [], opts)
     |> handle_response
   end
 
+  # This function will attempt to carry out a `find` operation. If that fails
+  # with a 429 (Too Many Requests), close.io will provide us rate limiting
+  # information. Use this to wait till the end of the reset window and retry.
+  # See https://developer.close.io/#ratelimits for more info
+  #
+  # REVIEW: I've only implemented this on `find` as that's the only place we're
+  # hitting rate limits and it limits the scope of this change (in case of
+  # fubar). Should this be applied more generally?
+  defp find_respecting_rate_limit(resource, search_term, opts) do
+    opts = merge_search_term_into_opts(search_term, opts)
+
+    response = find(resource, opts)
+
+    case response do
+      {:error, %{status_code: 429, body: %{"rate_reset" => rate_reset}}} ->
+        wait_and_retry(resource, opts, rate_reset)
+
+      no_rate_limit ->
+        no_rate_limit
+    end
+  end
+
+  defp wait_and_retry(resource, opts, rate_reset) do
```

Comment:
I think this should be extracted from the client to allow us to mock it in tests rather than switch behaviour based on environment

Reply:
Not sure I understand, are you referring to the decision on whether or not to sleep?

Reply:
@tpbowden - have pushed up a change extracting it to config. Look OK? Any thoughts on the `REVIEW` comments?